src/LiveComponent/src/EventListener/DeferLiveComponentSubscriber.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symfony\UX\LiveComponent\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\UX\LiveComponent\Util\LiveControllerAttributesCreator;
+use Symfony\UX\TwigComponent\ComponentStack;
+use Symfony\UX\TwigComponent\Event\PreRenderEvent;
+
+final class DeferLiveComponentSubscriber implements EventSubscriberInterface
+{
+    private const PRIORITY = 16;
+
+    private const DEFAULT_LOADING_TAG = 'div';
+
+    private const DEFAULT_LOADING_TEMPLATE = null;
+
+    public function __construct(
+        private ComponentStack $componentStack,
+        private LiveControllerAttributesCreator $attributesCreator,
+    ) {
+    }
+
+    public function onPreRender(PreRenderEvent $event): void
+    {
+        $mountedComponent = $event->getMountedComponent();
+        $inputProps = $mountedComponent->getInputProps();
+        $isDeferred = $inputProps['defer'] ?? false;
+
+        if (!$isDeferred) {
+            return;
+        }
+
+        $metadata = $event->getMetadata();
+        $variables = $event->getVariables();
+        $attributesKey = $metadata->getAttributesVar();
+
+        $event->setTemplate('@LiveComponent/deferred.html.twig');
+
+        $originalAttributes = $variables[$attributesKey]->all();
+
+        $attributes = $originalAttributes;
+        $attributes = array_filter(
+            $attributes,
+            fn ($key) => !\in_array($key, ['defer', 'loading-template', 'loading-tag'], true),
+            \ARRAY_FILTER_USE_KEY,
+        );
+
+        $variables[$attributesKey] = $attributes;
+        $variables['loadingTag'] = $originalAttributes['loading-tag'] ?? self::DEFAULT_LOADING_TAG;
+        $variables['loadingTemplate'] = $originalAttributes['loading-template'] ?? self::DEFAULT_LOADING_TEMPLATE;
+
+        $mountedComponent->setAttributes(
+            $mountedComponent->getAttributes()->without('defer', 'loading-template', 'loading-tag'),
+        );
```

Comment:
@kbond does this look clunky to you? The fact that we need to... kind of modify the attributes in `$variables[$attributesKey]` AND also on `MountedComponent`?

Should we be doing this in an earlier hook? Something that can intercept the `defer` input prop earlier and remove it? Maybe set that data onto `MountedComponent::extraMetadata()` so it can be read here to set the variables?

Reply:
Hmm, perhaps, and I agree it is clunky but it _does_ work within the current feature set. Opportunity for a future refactor.. I know I've said this for a few things recently. Maybe _ugly === ok_ until we need to duplicate it elsewhere?

Reply:
Yea, I definitely have that on my mind too - finding the balance. What about:

A) In `PostMountEvent` https://github.com/symfony/ux/blob/2.x/src/TwigComponent/src/ComponentFactory.php#L195 - we add a new `addExtraMetadata()` method to the event.

B) Then, we keep track of that and put it into the `MountedComponent` - https://github.com/symfony/ux/blob/2.x/src/TwigComponent/src/ComponentFactory.php#L91-L113 - a little odd, as we need to carry return that "extra metadata" from the private `postMount()`, but I can't think of a better way. The user can intercept and remove "extra keys" from `PostMount`... but they don't have access to "attach" that to the `MountedComponent`.

Reply:
Yeah, sounds like that would work. I guess I'd be a little concerned about splitting this logic into two places (assuming I am understanding correctly). It's clunky here but at least it's in one spot (which makes it easier to reason about imo).

Reply:
Gotcha. I'm leaning towards my solution. It just doesn't feel right to allow `defer` to get ALL the way into attributes... just for us to remove it later, because it is never meant to be an attribute. It is really a special "prop", which should be processed and removed before it ever gets into attributes.

Sorry for the back and forth @jakubtobiasz but does this make sense? It would allow us to remove the new method in `MountedComponent`

Reply:
> I'm leaning towards my solution

I'm ok with this - no strong objection.

Reply:
I'll try to take a look this week 🫡. I'll reach you once I find some difficulties 😂.